onnxruntime/core/common/threadpool.cc

```diff
@@ -57,6 +57,29 @@ void ThreadPool::ParallelFor(int32_t total, std::function<void(int32_t)> fn) {
   barrier.Wait();
 }
 
+void ThreadPool::BatchParallelFor(int32_t total, std::function<void(int32_t)> fn, int32_t batch_size) {
+  if (total <= 0)
+    return;
+
+  if (total == 1 || batch_size <= 1) {
+    fn(0);
+    return;
+  }
+
+  if (batch_size == total) {
+    ParallelFor(total, fn);
+    return;
+  }
+
```

Comment:
The behavior of this seems inconsistent. If the batch size is equal to the total, you're going to call fn on different threads each time. If batch size is not equal to total, you're going to make batch size calls to fn on a thread each time.

Reply:
I don't think there is any inconsistency here. if batch size is not equal to total, there is not going to be batch size calls to fn on a thread; it's either `floor(total/batch_size)` or `floor(total/batch_size)+1` calls on each thread